test/helpers.js

```diff
@@ -155,3 +159,79 @@ exports.cleanupPublish = function (app, options) {
     ], callback);
   };
 };
+
+// creates a new server to test against
+// takes starting npmrc copies to a tmp file
+// runs npm command w/ args against tmp file userconfig
+// - has blank globalconfig
+// npmrc files can use variables:
+// - $PORT : testing server port
+// - $TOKEN : the generatedToken
+const blank = path.join(__dirname, 'fixtures', 'npm-auth', 'blank.npmrc');
+exports.testNPM = function testNPM(registry, options) {
```

Comment:
can move to macro, this fn was ported from tap which has fail/end/teardown. We would still need to pass a special teardown fn in.